### DISCUSSION TITLE
Fetching json data from public directory during static site generation

### DISCUSSION THREAD

**Discussion by janusznoszczynski** [nuxt/nuxt#21063] [Questions]
Imagine project with single page:

_pages/index.vue_
```
<template>
    <p>{{ data }}</p>
</template>

<script setup>
const { data } = useFetch('http://localhost:3000/mydata.json');
</script>
```

_And static json file_
```
public/mydata.json
{
    "title": "Hello World"
}
```

The problem with is code is that after running:

1. npm run generate
2. npm run preview

Variable **data** will be null because request http://localhost:3000/mydata.json will fail.

Is there any way to fetch data from public folder during static site generation?

**Comment by maou-shonen** [comment]
1.
```diff
--- useFetch('http://localhost:3000/mydata.json');
+++ useFetch('/mydata.json');
```

2. use nuxt/content module

3. direct imports json
```ts
import mydata from '~/public/mydata.json'
```

**Reply by lustremedia** [reply]
useFetch('/mydata.json'); does not work in development whereas useFetch('http://localhost:3000/mydata.json'); does not work in prod using ssr  so 1. is really not an option ...
